src/generator.cs

```diff
@@ -4724,6 +4724,15 @@ void GenerateAsyncMethod (MemberInformation original_minfo, AsyncMethodKind asyn
 				ttype = "Tuple<bool,NSError>";
 		}
 		print ("var tcs = new TaskCompletionSource<{0}> ();", ttype);
+
+		// Intentionaly disable warning 0219 when method's return type is != void and AsyncMethodKind == Plain we ignore the returned value
+		bool shouldDisable219 = !is_void && asyncKind == AsyncMethodKind.Plain;
```

Comment:
Wouldn't it be easier to just not generate the `var result = ` part of the statement?